packages/@webex/webex-core/test/unit/spec/webex-core.js

```diff
@@ -169,6 +175,84 @@ describe('Webex', () => {
     });
   });
 
+  describe('initializes with interceptors', () => {
+    [
+      // 4 pre, 4 post, 9 remaining default = 17
+      [
+        'defaults to existing interceptors if undefined',
+        undefined,
+        17,
+        [
+          'RequestTimingInterceptor',
+          'RequestEventInterceptor',
+          'WebexTrackingIdInterceptor',
+          'RateLimitInterceptor',
+          'ServiceInterceptor',
+          'UserAgentInterceptor',
+          'WebexUserAgentInterceptor',
+          'AuthInterceptor',
+          'PayloadTransformerInterceptor',
+          'RedirectInterceptor',
+          'DefaultOptionsInterceptor',
+          'HostMapInterceptor',
+          'ServerErrorInterceptor',
+          'HttpStatusInterceptor',
+          'NetworkTimingInterceptor',
+          'EmbargoInterceptor',
+          'RateLimitInterceptor',
+        ],
+      ],
+      [
+        'only adds PayloadTransformerInterceptor',
+        {interceptors: {PayloadTransformerInterceptor: PayloadTransformerInterceptor.create}},
+        1,
+        ['PayloadTransformerInterceptor'],
+      ],
+      [
+        'adds multiple interceptors',
+        {
+          interceptors: {
+            PayloadTransformerInterceptor: PayloadTransformerInterceptor.create,
+            RequestTimingInterceptor: RequestTimingInterceptor.create,
+          },
+        },
+        2,
+        ['PayloadTransformerInterceptor', 'RequestTimingInterceptor'],
+      ],
+      [
+        'adds no interceptors',
+        {
+          interceptors: [],
+        },
+        0,
+        [],
+      ],
+    ].forEach(
+      ([message, interceptorsConfig, expectedNumberOfInterceptors, expectedInterceptorNames]) => {
+        let defaultOptions;
+        beforeEach(() => {
+          defaultOptions = {method: 'GET', url: 'https://fake-url.webex.com'};
+        });
+        it(message, async () => {
+          const webex = new WebexCore({
+            config: interceptorsConfig,
+          });
+          // don't care if it resolves or rejects, just need to see interceptors
+          try {
+            await webex.request(defaultOptions);
```

Comment:
i spent hours trying to stub out part of request, but couldn't get it to work correctly.